torch/csrc/jit/passes/onnx/cast_all_constant_to_floating.cpp

```diff
@@ -38,22 +40,23 @@ void CastAllConstantToFloating(Block* block) {
           case at::ScalarType::Short:
           case at::ScalarType::Bool:
             to_type = ATenTypeToOnnxType(val.scalar_type());
-            val = val.to(at::ScalarType::Float);
+            val_cast = val.to(at::ScalarType::Float);
             break;
 
           case at::ScalarType::Long:
             to_type = ATenTypeToOnnxType(val.scalar_type());
-            val = val.to(at::ScalarType::Double);
+            val_cast = val.to(at::ScalarType::Double);
             break;
 
           default:
             throw std::runtime_error("Unsupported types: complex, string");
         }
         // create a cast node
         node->removeAttribute(attr::value);
-        node->t_(attr::value, val);
+        node->t_(attr::value, val_cast);
         Node* cast_node = graph->create(onnx::Cast, 1);
         cast_node->i_(attr::to, to_type);
+        cast_node->output()->setType(TensorType::create(val));
```

Comment:
Is the creating of `val_cast` and `copy_` to avoid `TensorType::create(val)` being polluted? If so can we get the TensorType before cast and avoid the creating `val_cast` and `copy_`?

Reply:
done.